parsers/powershell.c

```diff
@@ -506,6 +529,38 @@ static bool parseFunction (tokenInfo *const token)
 	return readNext;
 }
 
+/* parse a class
+ *
+ * 	class MyClass {}
+ */
+static bool parseClass (tokenInfo *const token)
+{
+	bool readNext = true;
+	tokenInfo *nameFree = NULL;
+	const char *access;
+
+	readToken (token);
+
+	if (token->type != TOKEN_IDENTIFIER)
+		return false;
+
+	nameFree = newToken ();
+	copyToken (nameFree, token, true);
+	readToken (token);
+
+	makeClassTag (nameFree);
+
+	if (token->type == TOKEN_OPEN_CURLY)
```

Comment:
This may not work well for the input like:
```
class MyException : Exception {
```

If a class definition has {...} block ALWAYS, you can do something like:
```
while (token->type != TOKEN_OPEN_CURLY && token->type != TOKEN_EOF)
{
   readToken (token);
}

if (token->type != TOKEN_OPEN_CURLY)
  enterScope (token, nameFree->string, K_CLASS);
...
```

Reply:
There was an incorrect implementation. Thank you for pointing this out.
Please let me make an attempt to see if I can implement it in this PR.

Since the introduction of classes in PowerShell 5.0, class definitions always seem to include `{}`. Note that in PowerShell 5.1, the earliest currently supported version, the class syntax is:

```powershell
class <class-name> [: [<base-class>][,<interface-list>]] {
    [[<attribute>] [hidden] [static] <property-definition> ...]
    [<class-name>([<constructor-argument-list>])
      {<constructor-statement-list>} ...]
    [[<attribute>] [hidden] [static] <method-definition> ...]
}
```

* PowerShell 5.0 (not supported)
    * https://docs.microsoft.com/en-us/previous-versions/powershell/module/microsoft.powershell.core/about/about_classes?view=powershell-5.0
* PowerShell 5.1
    * https://docs.microsoft.com/ja-jp/powershell/module/microsoft.powershell.core/about/about_classes?view=powershell-5.1
* PowerShell 7.2
    * https://docs.microsoft.com/ja-jp/powershell/module/microsoft.powershell.core/about/about_classes?view=powershell-7.2

Reply:
>  Please let me make an attempt to see if I can implement it in this PR.

Yes, please try.

Reply:
Contrary to expectations, tags were created for variables in properties and methods within classes.

To keep this PR small and simple, no tags will be created for variables within classes in this PR.

Instead, TODO was left as a comment.

https://github.com/universal-ctags/ctags/pull/3476/commits/d45a7a453df569a5bfb5925182bd2a81b11e1926#diff-42a33750a868afb3e3d32b5f60a7bfe62fb32ee5697058cf74d9812bb5021c31R582